### PR TITLE
Refs #2525 - Exporting meshes with quads

### PR DESCRIPTION
**Pull request by cpetry**
Fixes:
* Exporting in general when quads being used in meshes
* Quads being processed and split into two triangles

Added an enum for clarification on gltfPrimitive modes. (optional)

### CODE REVIEW

**Comment by cpetry** [comment]
Refs #2525

File: Assets/UniGLTF/Runtime/SpringBoneJobs/InputPorts/FastSpringBoneBuffer.cs [hunk 1]
```diff
@@ -172,7 +172,14 @@ public void RestoreCurrentTails(NativeArray<Vector3> currentTails, NativeArray<V
...
                 {
                     // mark velocity zero
+#if UNITY_2022_2_OR_NEWER
                     currentTails.GetSubArray(offset, Logics.Length).AsSpan().Fill(new Vector3(float.NaN, float.NaN, float.NaN));
+#else
+                    var subArray = currentTails.GetSubArray(offset, Logics.Length);
+                    var value = new Vector3(float.NaN, float.NaN, float.NaN);
+                    for (int a = 0; a < subArray.Length; ++a)
+                        subArray[a] = value;
+#endif
```
Comment (ousttrue) [inline Assets/UniGLTF/Runtime/SpringBoneJobs/InputPorts/FastSpringBoneBuffer.cs:182]: This seems like unrelated code?

File: Assets/UniGLTF/Runtime/UniGLTF/IO/MeshIO/MeshExporter_SharedVertexBuffer.cs [hunk 2]
```diff
@@ -201,6 +156,119 @@ public static (glTFMesh, Dictionary<int, int> blendShapeIndexMap) Export(Exporti
...
+                {
+                    case MeshTopology.Triangles:
+                        if (submeshIndices.Length % 3 != 0)
+                            Debug.LogWarningFormat("triangle indices is not multiple of 3");
+                        GetTriangleIndices(indices, submeshIndices);
+                        break;
+                    case MeshTopology.Quads:
+                        if (submeshIndices.Length % 4 != 0)
+                            Debug.LogWarningFormat("quad indices is not multiple of 4");
+                        GetQuadIndices(indices, submeshIndices);
```
Comment (ousttrue) [inline Assets/UniGLTF/Runtime/UniGLTF/IO/MeshIO/MeshExporter_SharedVertexBuffer.cs:202]: Added quad mesh support 👍

**Review by ousttrue — APPROVED** [review]
LGTM
